Controller/CustomItem/SaveController.php

```diff
@@ -135,19 +135,22 @@ public function saveAction(int $objectId, ?int $itemId = null): Response
 
         $form->handleRequest($request);
 
+        $customItemId   = $customItem->getId();
+        $customItemName = $customItem->getName();
+
         if ($form->isValid()) {
-            $this->customItemModel->save($customItem);
+            $customItem = $this->customItemModel->save($customItem);
 
             if ($customItem->getChildCustomItem()) {
                 $customItem->getChildCustomItem()->generateNameForChildObject('contact', $contactId, $customItem);
-                $this->customItemModel->save($customItem->getChildCustomItem());
+                $customItem = $this->customItemModel->save($customItem->getChildCustomItem());
             }
 
             $this->flashBag->add(
                 $message,
                 [
-                    '%name%' => $customItem->getName(),
-                    '%url%'  => $this->routeProvider->buildEditRoute($objectId, $customItem->getId()),
+                    '%name%' => $customItemName,
+                    '%url%'  => $this->routeProvider->buildEditRoute($objectId, $customItemId),
```

Comment:
I'd suggest to revert this as the ID can be different (or more accurately, known) after the custom item is saved. Is there any reason why it was extracted before the save method?

Reply:
This was extracted because after save I was getting a null value on calling `$customItem->getId()`. This was causing failure in a few tests in `Tests/Unit/Controller/CustomItem/SaveControllerTest.php`. This was more of a hack to get the test to pass. Do you have any ideas to do it differently?

Reply:
Okay I just checked it The problem is not the code, it is the unit test. I'll have to modify it to accomodate these changes